### PR TITLE
IDEMPIERE-5665 - [Multi Selection] - Negate option in process parameter

### PR DESCRIPTION
**Pull request by jdalfre**
# Pull Request Checklist

- [X] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [X] My code follows the best practices of this project
- [X] I have performed a self-review of my own code
- [X] My code is easy to understand and review. 
- [X] I have checked my code and corrected any misspellings
- [X] In hard-to-understand areas, I have commented my code.
- [X] My changes generate no new warnings
### Tests
- [X] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/process/SvrProcess.java [hunk 1]
```diff
@@ -762,19 +762,21 @@ private void autoFillParameters(){
...
                 toField = map.containsKey(candidate + "_to") ? map.get(candidate + "_to") : null;
+                notField = map.containsKey(candidate + "_not") ? map.get(candidate + "_not") : null;
             }
 
             // try to match fields with same name as metadata declaration after stripping "_"
             if(field==null) {
             	String candidate = name.replace("_", "");
                 field = map.get(candidate);
-                toField = map.containsKey(candidate + "to") ? map.get(candidate + "to") : null;
+                notField = map.containsKey(candidate + "to") ? map.get(candidate + "to") : null;
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/process/SvrProcess.java:779]: this is wrong, should be the same as line 765 above, i.e keep toField and add notField for candidate + "not"

File: org.adempiere.base/src/org/compiere/process/SvrProcess.java [hunk 2]
```diff
@@ -788,6 +790,8 @@ private void autoFillParameters(){
                     	toField.set(this, parameter.getParameter_ToAsInt());
                 } else if (type.equals(String.class)) {
                     field.set(this, (String) parameter.getParameter());
+                    if(notField != null)
+                    	notField.set(this, (boolean) parameter.isNotClause());
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/process/SvrProcess.java:794]: it seems you are not passing the parameter value to the not field. can you provide an example of the annotation for this ?

**Review by hengsin — CHANGES_REQUESTED** [review]

File: org.adempiere.base/src/org/compiere/process/SvrProcess.java (hunk 1 above)
Comment (jdalfre) [reply org.adempiere.base/src/org/compiere/process/SvrProcess.java:779]: Yes, its Wrong @hengsin I will make the correction

File: org.adempiere.base/src/org/compiere/process/SvrProcess.java (hunk 2 above)
Comment (jdalfre) [reply org.adempiere.base/src/org/compiere/process/SvrProcess.java:794]: We can use annotations like https://wiki.idempiere.org/en/NF9_Annotated_Process_Parameters
An example:

@Parameter
   private String p_GL_Category_ID;
@Parameter
  private Boolean p_GL_Category_NOT;

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**